wagtail/wagtailcore/fields.py

```diff
@@ -12,31 +12,55 @@
 from wagtail.wagtailcore.blocks import Block, StreamBlock, StreamValue, BlockField
 
 
-class RichTextArea(WidgetWithScript, forms.Textarea):
+class BaseTextAreaWidget(WidgetWithScript, forms.Textarea):
+
+    def __init__(self, editor_config=None, **kwargs):
+        # set the base editor configuration if existent
+        self.editor_config = editor_config
+        super(BaseTextAreaWidget, self).__init__(**kwargs)
+
     def get_panel(self):
-        from wagtail.wagtailadmin.edit_handlers import RichTextFieldPanel
-        return RichTextFieldPanel
+        raise NotImplementedError(
+            "Class %s doesn't implement a required Method()" % (self.__class__.__name__))
 
     def render(self, name, value, attrs=None):
         if value is None:
             translated_value = None
         else:
             translated_value = expand_db_html(value, for_editor=True)
-        return super(RichTextArea, self).render(name, translated_value, attrs)
-
-    def render_js_init(self, id_, name, value):
-        return "makeRichTextEditable({0});".format(json.dumps(id_))
+        return super(BaseTextAreaWidget, self).render(name, translated_value, attrs)
 
     def value_from_datadict(self, data, files, name):
-        original_value = super(RichTextArea, self).value_from_datadict(data, files, name)
+        original_value = super(BaseTextAreaWidget, self).value_from_datadict(data, files, name)
         if original_value is None:
             return None
         return DbWhitelister.clean(original_value)
 
+    class Meta:
+        abstract = True
```

Comment:
Ahhh, I got myself confused. Class Meta is of course django Model specific and python, apart from the abc library, doesn't provide a strict abstract class definition. 

Sorry, I seem to be a bit thick today... I will update the codebase and add a test for the base class.